### DISCUSSION TITLE
Chrome discards and i3 workspaces

### DISCUSSION THREAD

**Discussion by yorrd** [i3/i3#5203] [Q&A]
Very specific issue. I'm probably Googling it wrong, help appreciated. The main goal here is to save battery since some Chrome tabs use a lot of CPU in idle and I can't control the website's CPU usage other than turning it off. Chromium has a neat `discard` feature for tabs and I force that behavior earlier than intended by Chrome with an extension.

- I want to discard tabs when I'm not using them. Let's specify: even if they're the only active tab in a window, I want chrome to discard them
- to discard tabs quicker, I'm using this modern and non-malware fork fo the great suspender: https://chrome.google.com/webstore/detail/the-great-er-discarder-er/plpkmjcnhhnpkblimgenmdhghfgghdpp/related
- the issue now is, windows in i3 on workspaces that are not currently shown on any monitor, seem to be detected as not-hidden by Chromium as per `about:discards`

Begging the question: can I tell Chromium that those windows are not actually visible? Shouldn't i3 do that in the case I described?

**Comment by Airblader** [comment]
It depends on what it means to be "marked as hidden". The logical way (to me) would be to check whether the window is mapped, and unmapping windows on inactive workspaces is exactly what i3 does.

It's also possible that it checks for `_NET_WM_STATE_HIDDEN`, which i3 does not set in this case, but only for windows within tabbed and stacked containers which are not currently focused within their parent (= invisible). Setting this atom for windows on inactive workspaces would violate the specification.

**Reply by yorrd** [reply]
Thanks for this information. While I don't really feel qualified, I'm gong to try and dig around within Chrome and their issue trackers, maybe I'll find relevant information. Helping a lot, appreciate it!
